server/app/services/apikey/ApiKeyService.java

```diff
@@ -156,7 +156,7 @@ public ApiKeyModel retireApiKey(Long apiKeyId, CiviFormProfile profile) {
     ApiKeyModel apiKey = maybeApiKey.get();
 
     if (apiKey.isRetired()) {
-      throw new NotChangeableException(String.format("ApiKey %s is already retired", apiKey));
+      throw new NotChangeableException(String.format("ApiKey %s is already retired", apiKey.id));
```

Comment:
Retired or not I didn't like that it was including the full record's data in the exception which was getting put in the logs. Especially the IP addresses on the allow list.